packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -272,10 +274,20 @@ const ProductDetail = () => {
             einstein.sendViewProduct(product)
             const childrenProducts = product.setProducts
             childrenProducts.map((child) => {
-                einstein.sendViewProduct(child)
+                try {
+                    einstein.sendViewProduct(child)
+                } catch (err) {
+                    console.error(err)
+                }
```

Comment:
```suggestion
                einstein.sendViewProduct(child)
```
Einstein catches its own errors, so we shouldn't need to wrap it.

Reply:
While testing, I encountered few times an issue where Einstein error was thrown and no active data tracking call happened, so keeping this as is.